generators/server/index.js

```diff
@@ -1024,6 +1024,31 @@ module.exports = JhipsterServerGenerator.extend({
                     this.copy(SERVER_MAIN_RES_DIR + 'config/cql/changelog/_insert_default_users.cql', SERVER_MAIN_RES_DIR + 'config/cql/changelog/00000000000000_insert_default_users.cql');
                 }
             }
+
+            if (this.applicationType === 'uaa') {
+                const keyStoreFile = SERVER_MAIN_RES_DIR + 'keystore.jks';
+                if (this.fs.exists(keyStoreFile)) {
+                    this.log(chalk.cyan('\nKeyStore \'' + keyStoreFile + '\' already exists. Leaving unchanged.\n'));
+                } else {
+                    shelljs.mkdir('-p', SERVER_MAIN_RES_DIR);
+                    var exitCode = shelljs.exec('keytool '+
+                        '-genkey ' +
+                        '-noprompt ' +
+                        '-keyalg RSA ' +
+                        '-alias selfsigned ' +
+                        '-keystore ' + keyStoreFile + ' ' +
+                        '-storepass password ' +
+                        '-keypass password ' +
+                        '-keysize 2048 ' +
+                        '-dname "CN=Java Hipster, OU=Development, O=' + this.packageName + ', L=, ST=, C="'
+                    ).code;
+                    if (exitCode !== 0) {
+                        this.log(chalk.red(`\nFailed to create a KeyStore with \'keytool\'`));
+                        shelljs.exit(exitCode);
```

Comment:
does this exit the generator itself? if so its better we use `this.error()` to avoid any process being affected